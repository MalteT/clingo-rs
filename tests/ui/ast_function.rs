use clingo::*;

fn main() {
    let name = String::from("name");
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = ast::Term::from(sym);
    let term2 = ast::Term::from(sym);
    let args = vec![term1,term2];
    let fun = ast::Function::new(&name, &args).unwrap();
    drop(args);
    println!("{:?}",fun);
}