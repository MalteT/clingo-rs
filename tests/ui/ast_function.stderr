error[E0505]: cannot move out of `name` because it is borrowed
  --> $DIR/ast_function.rs:10:10
   |
9  |     let fun = ast::Function::new(&name, &args).unwrap();
   |                                  ----- borrow of `name` occurs here
10 |     drop(name);
   |          ^^^^ move out of `name` occurs here
11 |     drop(args);
12 |     println!("{:?}",fun);
   |                     --- borrow later used here

error[E0505]: cannot move out of `args` because it is borrowed
  --> $DIR/ast_function.rs:11:10
   |
9  |     let fun = ast::Function::new(&name, &args).unwrap();
   |                                         ----- borrow of `args` occurs here
10 |     drop(name);
11 |     drop(args);
   |          ^^^^ move out of `args` occurs here
12 |     println!("{:?}",fun);
   |                     --- borrow later used here
